qp_klp/process_amplicon_job.py

```diff
@@ -306,9 +306,15 @@ def process_amplicon(mapping_file_path, qclient, run_identifier, out_dir,
         for study_id in gpf_job.prep_file_paths:
             for prep_file_path in gpf_job.prep_file_paths[study_id]:
                 metadata = pd.read_csv(prep_file_path,
+                                       dtype=str,
                                        delimiter='\t',
-                                       index_col='sample_name').to_dict(
-                    'index')
+                                       index_col='sample_name')
```

Comment:
`index_col` does not interact as expected with `dtype`. This will still incur a type conversion. Converting it to `str` after the fact is lossy. Please set remove, and set an explicit index with `set_index` after parse

```python
>>> df = pd.read_csv('clinical_metadata_formatted.tsv', sep='\t', dtype=str, index_col='sample-id')
>>> df.index[:5]
Float64Index([10317.000106498, 10317.000107092, 10317.000106499,
              10317.000107093, 10317.000106684],
             dtype='float64', name='sample-id')
```

Reply:
@wasade I can see it - it didn't happen in the case that spawned this hot fix but in general if the index column gets interpreted as a numeric directly instead of getting casted to string first with the other columns before then being converted when it becomes the index, you would get potential data loss.

Reply:
I think we're in agreement. Please do not allow the index column to be cast to numeric.